pinot-tools/src/main/java/org/apache/pinot/tools/admin/PinotAdministrator.java

```diff
@@ -140,56 +142,71 @@ public class PinotAdministrator {
     SUBCOMMAND_MAP.put("FileSystem", new FileSystemCommand());
   }
 
-  @CommandLine.Option(names = {"-help", "-h", "--h", "--help"}, required = false,
-      description = "Print this message.")
+  @CommandLine.Option(names = {"-help", "-h", "--h", "--help"}, required = false, description = "Print this message.")
   private boolean _help = false;
 
-  @CommandLine.Option(names = {"-version", "-v", "--v", "--version"}, required = false,
-      description = "Print the version of Pinot package.")
+  @CommandLine.Option(names = {"-version", "-v", "--v", "--version"}, required = false, description = "Print the "
+                                                                                                      + "version of "
+                                                                                                      + "Pinot "
+                                                                                                      + "package.")
   private boolean _version = false;
 
   private int _status = 1;
 
+  private final Writer _outWriter;
+
+  private final Versions _versions;
+
+  public PinotAdministrator(Writer outWriter, Versions versions) {
+    _outWriter = outWriter;
+    _versions = versions;
+  }
+
   public void execute(String[] args) {
     try {
-      CommandLine commandLine = new CommandLine(this);
-      for (Map.Entry<String, Command> subCommand : this.getSubCommands().entrySet()) {
-        commandLine.addSubcommand(subCommand.getKey(), subCommand.getValue());
-      }
-      CommandLine.ParseResult parseResult = commandLine.parseArgs(args);
-      // TODO: Use the natively supported version and usage by picocli
-      // see https://picocli.info/#_mixin_standard_help_options
-      if (!parseResult.hasSubcommand()) {
-        if (_version) {
-          printVersion();
-          _status = 0;
-        } else if (_help) {
-          printUsage();
-          _status = 0;
-        }
-      } else {
+      var commandLine = new CommandLine(this);
```

Comment:
seems like we can't support var b/c of legacy JDK8 support :-(

Reply:
I wasn't aware my apologies. I removed all JDK8+ specific things.

Reply:
no worries. i wasn't aware either. thank you for quick turn around!